### ISSUE TITLE
Tuesday seems too soon

### ISSUE THREAD

**Issue by chadmart** [usc-csci201-fall2013/simcity201#48]
There is less than a week left till a huge chunk of our overall grade is due, and more than a week of coding to left to do. We could use a few extra days to get all of the norms running and tested. Other classes and sleep are starting to get in the way of us completing the project on time. Is there any chance of extending the due date to the following Sunday? The extra time could mean the difference between someone getting an A in the class versus a C.

@dwilczyn @crowley1958 @alghanmi

**Comment by smoran02** [comment]
I agree, I have a big project like this to do in several classes and am not sure how I will have the time to complete them all. I think this the scope of this assignment is doable before Tuesday, but not if I'm going to do the work for my other classes too.

**Comment by jlazzaridean** [comment]
I also agree (along with team 37). Getting all normative scenarios running, along with the gui components and full unit testing of all agents, will take longer than a few days.

**Comment by waynchi** [comment]
I feel like our group is on track excluding Gui. Given the fact that we
never took CS200 (And the fact that A\* was just added back as part of the
requirements when it wasn't a part of the requirements before), I can't
give any estimate on whether or not our group is on track for that task.

**Comment by dhakiki** [comment]
Our team has been meeting many times a week. We have our roles coded and tested individually but integrating requires way more testing and development and could take up until the end of the weekend, leaving insufficient time for animations and gui (and, considering we get through those, a\* implementation).

**Comment by Cherys** [comment]
There is no way my team has time to put our best work out there. It will be rushed, haphazard, and probably incomplete in order to get as much done as possible by this deadline

**Comment by chadmart** [comment]
Even if this extension is granted, I don't know if my team could get a working city. There have been so many issues with integrating roles into 1 restaurant. The design was rushed, and there are a lot of concurrency issues such as deadlock. I feel we are all going to fail except for teams that have people who have experience writing these complex systems. The factory project only had 6 agents. This system has at least 50. Debugging and testing seems impossible. We were not prepared for this task. We feel lost.

**Comment by Cherys** [comment]
I'm glad someone said what we've all been feeling. The designs WERE rushed, and far too small and simple for the enormous project at hand. We STILL don't know exactly what the requirements are, and back when we made our designs many of the new requirements weren't known. We are currently working on two giant projects actually--one is the creation of the city, and the other is a serious overhaul of our restaurants. People whose restaurants were glitchy are going to inadvertently entrap agents that need to be elsewhere in the city, the guis and input panels have to be completely reformatted to fit the generalized view screen, interactions with the market have to be redone, and all agents have to be redefined and reworked. Not only that, but we have to ADD functionality on top of all that. That assignment by itself is undoubtably the size of one of our previous two-week assignments. That's without even considering the city which we have had to build from scratch with vague/changing requirements, difficulty scheduling meeting times, and some team members with more or less ability and skill than others (and those with more don't have time to help those with less because of the crushing scope of this project). This is all happening during the time where ALL classes--not just CSCI 201--are increasing their required workloads and assigning large final projects. There is no _possible_ way that this project is going to get done well or completely by more than (perhaps) a small fraction of students, and those students are either not taking any other classes, or they significantly sacrificed their grades in their other classes to struggle with this one. And, as of the moment, we have no choice but to sacrifice our other classes as much as we can possibly afford, because taking any time away from this project to sleep or work on other classes practically guarantees that we will fail this course.

We are getting desperate
